10/22/2025 22:08:13 - INFO - __main__ - ***** Training arguments *****
10/22/2025 22:08:13 - INFO - __main__ - Namespace(config='configs/ddpm.yaml', data_dir='/ocean/projects/cis250019p/jye9/11685-project/data', is_cifar_10=True, image_size=32, batch_size=64, num_workers=4, num_classes=10, run_name='exp-0-ddpm', output_dir='/ocean/projects/cis250019p/jye9/11685-project/experiments', num_epochs=300, learning_rate=0.0005, weight_decay=0.0001, grad_clip=1.0, seed=42, mixed_precision='none', use_val=True, num_train_timesteps=1000, num_inference_steps=250, beta_start=0.0001, beta_end=0.02, beta_schedule='cosine', variance_type='fixed_small', prediction_type='epsilon', clip_sample=True, clip_sample_range=1.0, optimizer_type='AdamW', scheduler_type='ConsineAnnealingLR', unet_in_size=32, unet_in_ch=3, unet_ch=128, unet_ch_mult=[1, 2, 2, 4], unet_attn=[1, 2, 3], unet_num_res_blocks=2, unet_dropout=0.1, latent_ddpm=False, pretrained_vae='/ocean/projects/cis250019p/jye9/11685-project/pretrained/model.ckpt', use_cfg=False, cfg_guidance_scale=2.0, use_ddim=False, ckpt=None, device='cuda', rank=0, world_size=1, predictor_type='epsilon', distributed=False, local_rank=0, total_batch_size=64, max_train_steps=234300)
10/22/2025 22:08:13 - INFO - __main__ - ***** Running training *****
10/22/2025 22:08:13 - INFO - __main__ -   Num examples = 50000
10/22/2025 22:08:13 - INFO - __main__ -   Num Epochs = 300
10/22/2025 22:08:13 - INFO - __main__ -   Instantaneous batch size per device = 64
10/22/2025 22:08:13 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 64
10/22/2025 22:08:13 - INFO - __main__ -   Total optimization steps per epoch 781
10/22/2025 22:08:13 - INFO - __main__ -   Total optimization steps = 234300
  0%|                                                                                                                                 | 0/234300 [00:00<?, ?it/s]10/22/2025 22:08:13 - INFO - __main__ - Epoch 1/300
  0%|                                                                                                                  | 1/234300 [02:37<10239:00:19, 157.32s/it]10/22/2025 22:10:50 - INFO - __main__ - Epoch 1/300, Step 780/781, Loss 0.0763448029756546 (0.13666751259095997)
10/22/2025 22:10:56 - INFO - __main__ - Epoch 1/300 Validation Loss: 0.065691
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 250/250 [00:03<00:00, 65.83it/s]
10/22/2025 22:11:00 - INFO - numexpr.utils - Note: NumExpr detected 40 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 16.3<00:00, 67.03it/s]
10/22/2025 22:11:00 - INFO - numexpr.utils - NumExpr defaulting to 16 threads.
Traceback (most recent call last):
  File "/ocean/projects/cis250019p/jye9/conda/envs/11685/lib/python3.12/site-packages/torch/serialization.py", line 967, in save
    _save(
  File "/ocean/projects/cis250019p/jye9/conda/envs/11685/lib/python3.12/site-packages/torch/serialization.py", line 1268, in _save
    zip_file.write_record(name, storage, num_bytes)
RuntimeError: [enforce fail at inline_container.cc:858] . PytorchStreamWriter failed writing file data/1067: file write failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/jet/home/jye9/Project/11685_project/train.py", line 596, in <module>
    main()
  File "/jet/home/jye9/Project/11685_project/train.py", line 592, in main
    save_checkpoint(unet_wo_ddp, scheduler_wo_ddp, vae_wo_ddp, class_embedder, optimizer, epoch, save_dir=save_dir)
  File "/jet/home/jye9/Project/11685_project/utils/checkpoint.py", line 50, in save_checkpoint
    torch.save(checkpoint, checkpoint_path)
  File "/ocean/projects/cis250019p/jye9/conda/envs/11685/lib/python3.12/site-packages/torch/serialization.py", line 966, in save
    with _open_zipfile_writer(f) as opened_zipfile:
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/ocean/projects/cis250019p/jye9/conda/envs/11685/lib/python3.12/site-packages/torch/serialization.py", line 798, in __exit__
    self.file_like.write_end_of_file()
RuntimeError: [enforce fail at inline_container.cc:664] . unexpected pos 636337920 vs 636337808
